DPG/Tasks/qaEventTrackLite.cxx

```diff
@@ -578,18 +578,38 @@ struct qaEventTrackLite {
     return true;
   }
 
-  bool isPdgSelected(const Int_t pdgcode)
+  bool isPdgSelected(const int pdgcode)
   { // mimics selection of charged particles or id particles
-    Int_t abspdgcode = TMath::Abs(pdgcode);
-    if (abspdgcode == pdgCodeSel)
+
+    if (pdgcode == pdgCodeSel) { // Check that the pdg code is exactly what was asked
       return true;
+    }
+    const int abspdgcode = abs(pdgcode);
     if (pdgCodeSel == 1 || pdgCodeSel == 2) {
-      if (abspdgcode == 211 || abspdgcode == 321 || abspdgcode == 2212 || abspdgcode == 11 || abspdgcode == 13)
-        return true;
-      if (pdgCodeSel == 2) {
-        if (abspdgcode == 3222 || abspdgcode == 3112 || abspdgcode == 3312 || abspdgcode == 3334 || abspdgcode == 1000010020 || abspdgcode == 1000010030 || abspdgcode == 1000020030 || abspdgcode == 1000020040 || abspdgcode == 1010010030 || abspdgcode == 1010020040)
+      switch (abspdgcode) {
```

Comment:
my ignorance: are you sure that this `switch` statement does the same thing of the original `if`? Reading this code I naively interpret that `return true;` is done only for `case 2212`

Reply:
it does the same, but now it's more clear, I also removed the abs value for the first selection